contrib/grimshot

```diff
@@ -28,7 +28,7 @@ fi
 
 ACTION=${1:-usage}
 SUBJECT=${2:-screen}
-FILE=${3:-$(getTargetDirectory)/$(date -Ins).png}
+FILE=${3:-$(getTargetDirectory)/$(date -Ins 2>/dev/null || date +%Y%m%d_%Hh%Mm%Ss_grim).png}
```

Comment:
This generates a very funny-looking filename and I don't see why files should be named `grim`.

```console
$ date +%Y%m%d_%Hh%Mm%Ss_grim
20201226_17h09m53s_grim
```

This also has the issue of overwriting consecutive screenshots taken within one second.

```suggestion
FILE=${3:-$(getTargetDirectory)/$(date +%Y%m%dT%H%M%S.%N).png}
```

I think this should work fine on all platforms, and avoid having diverging / per-platform variations. This looks like:

> 20201226T160853.228348830

Reply:
`%N` is a GNU extension not supported by `strftime` anywhere but a special-cased in GNU date. See strftime in [POSIX](https://pubs.opengroup.org/onlinepubs/9699919799/functions/strftime.html), [Linux](https://www.man7.org/linux/man-pages/man3/strftime.3.html), [FreeBSD](https://man.freebsd.org/strftime/3), [DragonFly](https://www.dragonflybsd.org/cgi/web-man?command=strftime&section=3), [NetBSD](https://man.netbsd.org/strftime.3), [OpenBSD](https://man.openbsd.org/strftime.3).

Reply:
Oh, you're right, my bad.

Is there no way to generate a timestamp _with_ milliseconds? Taking more than one screenshot in under one second is a pretty valid use-case, and I'd rather not break that.

I do agree that `grimshot` should be compatible with POSIX's `date` though.